closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1115,82 +1524,130 @@ private void listAvailableProfiles() {
-                    parseProperty(securityProps.getProperty(profileToPrint + ".securerandom.algorithm")));
-            System.out.println();
+        /**
+         * Only set a property if the value is not null.
+         *
+         * @param property      the property to be set
+         * @param propertyKey   the property key in the java.security file
+         * @return              whether the property was set
+         */
+        private boolean setProperty(String property, String propertyKey, List<String> allInfo) {
+            if (debug != null) {
+                debug.println("Setting property: " + property);
+            }
+            String value = securityProps.getProperty(propertyKey);
+            value = parseProperty(value);
+            String newValue = null;
+            if (value != null) {
+                // Add to info to create hash.
+                allInfo.add(propertyKey + "=" + value);
+
+                // Check if property overrides, adds to or removes from previous value.
+                String existingValue = profileProperties.get(property);
+                if (!value.isBlank() && value.startsWith("+")) {
+                    if (isPropertyAppendable(property)) {
+                        // Append additional values to property.
+                        value = value.substring(1, value.length()).trim();
+
+                        // Take existing value of property into account, if applicable.
+                        if (existingValue == null) {
+                            printStackTraceAndExit("Property '" + property + "' does not exist in parent profile. Cannot append.");
+                        } else if (existingValue.isBlank()) {
+                            newValue = value;
+                        } else {
+                            newValue = (value.isBlank()) ? existingValue : existingValue + ", " + value;
+                        }
+                    } else {
+                        printStackTraceAndExit("Property '" + property + "' is not appendable.");
+                    }
+                } else if (!value.isBlank() && value.startsWith("-")) {
+                    if (isPropertyAppendable(property)) {
+                        // Remove values from property.
+                        value = value.substring(1, value.length()).trim();
+                        if (!value.isBlank()) {
+                            List<String> existingValues = Arrays.asList(existingValue.split(","));
+                            existingValues = existingValues.stream().map(v -> v.trim()).collect(Collectors.toList());
+                            List<String> valuesToRemove = Arrays.asList(value.split(","));
+                            for (String valueToRemove : valuesToRemove) {
+                                if (existingValues.contains(valueToRemove.trim())) {
+                                    existingValues.remove(valueToRemove.trim());
```

Comment:
`remove()` will answer whether the list changed:
```java
                                if (!existingValues.remove(valueToRemove.trim())) {
                                    printStackTraceAndExit("Value '" + valueToRemove + "' is not in existing values.");
                                }
```

Reply:
Changed to suggestion.